src/modelfree/common/utils.py

```diff
@@ -38,3 +38,17 @@ def add_artifacts(run, dirname, ingredient=None):
             relroot = os.path.relpath(path, dirname)
             name = prefix + relroot.replace('/', '_') + '_' + file
             run.add_artifact(path, name=name)
+
+
+def dict_deep_copy(d):
+    """Perform a deep copy on nested dictionaries.
+
+    Performs a shallow copy using `dict()`. Note this will *lose* subclasses,
```

Comment:
This documentation is a bit confusing between shallow and deep, and it doesn't look like we're actually using the dict conversion function anywhere in here

Reply:
Thanks, you're right this is confusing, addressed.